### PR TITLE
distsql: add simple prometheus metric

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: distsql/distsql.go [hunk 1]
```diff
@@ -260,14 +261,29 @@ func (pr *partialResult) Close() error {
...
+		} else {
+			queryCounter.WithLabelValues(querySucc).Inc()
+		}
+	}()
+
 	// Convert tipb.*Request to kv.Request.
-	kvReq, err := composeRequest(req, keyRanges, concurrency, keepOrder)
-	if err != nil {
-		return nil, errors.Trace(err)
+	kvReq, e := composeRequest(req, keyRanges, concurrency, keepOrder)
```
Comment (shenli) [inline distsql/distsql.go]: We always use err/err1....errx to represent error.

File: distsql/metrics.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+var (
+	queryHistgram = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: "tidb",
+			Subsystem: "distsql",
+			Name:      "handle_query_duration_seconds",
+			Help:      "Bucketed histogram of processing time (s) of handled queries.",
+			Buckets:   prometheus.ExponentialBuckets(0.0005, 2, 13),
```
Comment (shenli) [inline distsql/metrics.go]: @siddontang Please review this parameter.

Comment (siddontang) [inline distsql/metrics.go]: LGTM

**Comment by shenli** [comment]
Rest LGTM

**Comment by shenli** [comment]
LGTM
@coocood @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**